services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -209,48 +233,53 @@ async def _start_dynamic_service(
     user_id: UserID,
     project_uuid: ProjectID,
     node_uuid: NodeID,
-):
+) -> None:
     if not _is_node_dynamic(service_key):
         return
-    project_running_nodes = await director_v2_api.list_dynamic_services(
-        request.app, user_id, f"{project_uuid}"
-    )
 
+    # this is a dynamic node, let's gather its resources and start it
+
+    lock_key = _get_service_start_lock_key(user_id, project_uuid)
+    client_sdk = get_redis_lock_manager_client_sdk(request.app)
     project_settings = get_settings(request.app).WEBSERVER_PROJECTS
-    assert project_settings  # nosec
-    if project_settings.PROJECTS_MAX_NUM_RUNNING_DYNAMIC_NODES > 0 and (
-        len(project_running_nodes)
-        >= project_settings.PROJECTS_MAX_NUM_RUNNING_DYNAMIC_NODES
+
+    async with client_sdk.lock_context(
+        lock_key,
+        blocking=True,
+        blocking_timeout_s=project_settings.total_project_dynamic_nodes_creation_interval,
```

Comment:
you need to assert it exists. Pylance configuration? can we check this specific one together?